modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -55,6 +55,7 @@
     public static final String DATE_LIBRARY = "dateLibrary";
     public static final String JAVA8_MODE = "java8";
     public static final String SUPPORT_ASYNC = "supportAsync";
+    public static final String WITH_AWSV4_SIGNATURE = "withAWSV4Signature";
```

Comment:
I thought we were going to use something more general like `detectAmznApiGatewayExtensions`?

Are you doing this because the golang variant already uses this switch?? OK, I guess... we also all agreed that the golang variant was garbage, right? The fact that it had an AWS region hardcoded for signing is a good indicator that we probably should not be following that person's lead.

Anyway - OK to leave this as-is for now if necessary. We can always generalize it later before commiting to the main project.